### PR TITLE
NIFI-9586 - investigate CI failures

### PR DESCRIPTION
**Pull request by greyp9**
#### Description of PR

- poll CI machine before and after build for disk space availability
- preserve surefire test output on build failure for analysis (retain for three days)

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [X] Have you verified that the full build is successful on JDK 8?
- [X] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -93,6 +94,17 @@ jobs:
...
           ${{ env.MAVEN_PROJECTS }}
+      - name: upload-on-failure
+        uses: actions/upload-artifact@v2
+        with:
+          name: surefire-reports-ubuntu-en
+          path: |
+            ./**/target/surefire-reports/*.txt
+            ./**/target/surefire-reports/*.xml
+          retention-days: 3
+        if: failure()
```
Comment (exceptionfactory) [inline .github/workflows/ci-workflow.yml:105]: Unfortunately it looks like this does not pull any files for the current failure on JDK 11, so perhaps capturing some other output may be helpful.

File: .github/workflows/ci-workflow.yml [hunk 2]
```diff
@@ -93,6 +94,17 @@ jobs:
...
+      - name: System Information (after)
```
Comment (exceptionfactory) [inline .github/workflows/ci-workflow.yml:106]: It is probably worth renaming this to Disk Usage or something similar.

File: .github/workflows/ci-workflow.yml [hunk 3]
```diff
@@ -191,3 +216,14 @@ jobs:
...
+        uses: actions/upload-artifact@v2
+        with:
+          name: surefire-reports-windows-fr
+          path: |
+            ./**/target/surefire-reports/*.txt
+            ./**/target/surefire-reports/*.xml
+          retention-days: 3
+        if: failure()
+      - name: System Information (after)
+        run: df
```
Comment (exceptionfactory) [inline .github/workflows/ci-workflow.yml:229]: Will this command work on Windows?

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for working on this issue @greyp9! It looks like these changes do not capture details for current failures on JDK 11, so it seems like a different approach is necessary.

File: .github/workflows/ci-workflow.yml (hunk 1 above)
Comment (greyp9) [reply .github/workflows/ci-workflow.yml:105]: This is actually good.  Now we know that there is no issue with the test itself, but rather with the maven invocation of surefire.

File: .github/workflows/ci-workflow.yml (hunk 2 above)
Comment (greyp9) [reply .github/workflows/ci-workflow.yml:106]: I chose this for symmetry with the preceding "System Information" label, which collects data before the build.  My thought was that there might be other non-disk information we might also want to collect after the build.  But open to a different name.

**Review by exceptionfactory — APPROVED** [review]
Thanks for the improvements @greyp9! It looks like the build completed on all platforms using the default LegacyForkNodeFactory in the Maven Surefire Plugin 3.0.0-M5. +1 Merging!

**Comment by joewitt** [comment]
All the goodness.  Thanks Paul and David!

**Pull request closed without merging**